torch/distributed/device_mesh.py

```diff
@@ -117,6 +117,20 @@ def create_sub_mesh(
 
             return res_submesh
 
+        def is_flatten_mesh(self, device_mesh: "DeviceMesh") -> bool:
+            root_mesh = self.get_root_mesh(device_mesh)
+            if not device_mesh.mesh_dim_names or len(device_mesh.mesh_dim_names) != 1:
+                return False
+            else:
+                flatten_mesh_dim_name = device_mesh.mesh_dim_names[0]
+                return (
+                    root_mesh in self.root_to_flatten_mapping.keys()
+                    and flatten_mesh_dim_name
+                    in self.root_to_flatten_mapping[root_mesh].keys()
+                    and self.root_to_flatten_mapping[root_mesh][flatten_mesh_dim_name]
+                    == device_mesh
```

Comment:
When can the third condition happen, `self.root_to_flatten_mapping[root_mesh][flatten_mesh_dim_name]== device_mesh`? I'm actually not sure why this can happen. Should we have an assert if that happens?

Reply:
I'm also wondering should we create a helper, `def get_flatten_mesh(root_mesh, flatten_mesh_name):`, given that we may be using this API to retrieve a flatten mesh back.

Reply:
> When can the third condition happen, `self.root_to_flatten_mapping[root_mesh][flatten_mesh_dim_name]== device_mesh`? I'm actually not sure why this can happen. Should we have an assert if that happens?

If the device_mesh passed in is the flatten mesh ,the third condition would happen. The mapping goes like this:
```
{
    "root_mesh's DeviceMesh":  {
           "dp_cp": flatten mesh's DeviceMesh
     }
}
```